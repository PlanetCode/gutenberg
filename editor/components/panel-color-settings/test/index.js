/**
 * External dependencies
 */
import { shallow } from 'enzyme';
import { noop } from 'lodash';

/**
 * Internal dependencies
 */
import { PanelColorSettings } from '../';

describe( 'PanelColorSettings', () => {
	it( 'matches the snapshot', () => {
		const wrapper = shallow(
			<PanelColorSettings
				title="Test Title"
				colors={ [] }
				backgroundColorProps={ {
					value: '#000',
					onChange: noop,
				} }
				textColorProps={ {
					value: '#111',
					onChange: noop,
				} }
				contrastCheckerProps={ {
					isLargeText: false,
					textColor: '#111',
					backgroundColor: '#000',
					fallbackTextColor: '#000',
					fallbackBackgroundColor: '#fff',
				} }
			/>
		);

		expect( wrapper ).toMatchSnapshot();
	} );
} );
