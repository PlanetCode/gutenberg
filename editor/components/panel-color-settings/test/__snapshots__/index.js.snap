// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelColorSettings matches the snapshot 1`] = `
<PanelBody
  className="editor-panel-color-settings"
  title={
    <span
      className="editor-panel-color-settings__panel-title"
    >
      Test Title
      <ColorArea
        ariaLabel="(current background color: #000)"
        colorValue="#000"
      />
      <ColorArea
        ariaLabel="(current text color: #111)"
        colorValue="#111"
      />
    </span>
  }
>
  <BaseControl
    label={
      <span
        className="editor-panel-color-settings__control-title"
      >
        Background Color
        <ColorArea
          ariaLabel="(current background color: #000)"
          colorValue="#000"
        />
      </span>
    }
  >
    <WithColorContext(ColorPalette)
      className="editor-panel-color-settings__color-palette"
      onChange={[Function]}
      value="#000"
    />
  </BaseControl>
  <BaseControl
    label={
      <span
        className="editor-panel-color-settings__control-title"
      >
        Text Color
        <ColorArea
          ariaLabel="(current text color: #111)"
          colorValue="#111"
        />
      </span>
    }
  >
    <WithColorContext(ColorPalette)
      className="editor-panel-color-settings__color-palette"
      onChange={[Function]}
      value="#111"
    />
  </BaseControl>
  <ContrastChecker
    backgroundColor="#000"
    fallbackBackgroundColor="#fff"
    fallbackTextColor="#000"
    isLargeText={false}
    textColor="#111"
  />
</PanelBody>
`;
